File: Cap 10/Topico 10.6/Topico 10.6.2/calcfat2.py
'''
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.6.2
 * 
 * Programa ..: PASSAGEM DE PARAMETRO POR REFERENCIA
'''

def FATORIAL(N, FAT):
  FAT[0] = 1
  for I in range(1, N+1):
    FAT[0] = FAT[0] * I

LIMITE = 0
RESP = [1]

print("CALCULO DE FATORIAL")
print("PROCEDIMENTO COM PASSAGEM DE PARAMETRO POR REFERENCIA")
print()
LIMITE = int(input("Qual fatorial (1-20): "))
print()
FATORIAL(LIMITE, RESP)
print("Fatorial =", RESP[0])

File: Cap 10/Topico 10.6/Topico 10.6.2/calcfat2.js
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 10 - Topico 10.6.2
 * 
 * Programa ..: PASSAGEM DE PARAMETRO POR REFERENCIA
 */

const readline = require('readline-sync');

function fatorial(N, FAT) {
  let I;
  FAT.value = 1;
  for (I = 1; I <= N; I++) {
    FAT.value *= I;
  }
}

let LIMITE;
let RESP = { value: 1 };
console.log("CALCULO DE FATORIAL");
console.log("PROCEDIMENTO COM PASSAGEM DE PARAMETRO POR REFERENCIA\n");
LIMITE = parseInt(readline.question("Qual fatorial (1-21): "));
console.log();
fatorial(LIMITE, RESP);
console.log("Fatorial = " + RESP.value);
